[tituswe] iP

tituswe opened the pull request:
### Saturday

> "nananananana" - My Chemical Romance

Implemented improvements to Saturday. Application now supports a comprehensive system which allows for:

- Adding tasks
- Managing tasks

To test Saturday _v0.1_: 

1. [download the application here](https://github.com/tituswe/ip/releases/tag/A-Jar).
2. double-click it
3. add your tasks
4. let it manage your tasks for you!

**Features**:
- [x] Add tasks
- [x] Delete tasks
- [x] Search for tasks
- [ ] **AI-powered** task completion suggestion system 👍 

Here's the `main` method:
```ruby
public static void main(String[] args) {
        String filePath = Storage.getFilePath();
        new Saturday(filePath).run();
    }
```

Commit:
Add Gradle support

tituswe committed:
Implemented Level 0. Greet

tituswe committed:
Implemented Level-1. Greet, Echo, Exit

tituswe committed:
Implemented Level-2. Add, List

tituswe committed:
Implemented TaskList and Task classes

tituswe committed:
Implemented Level-3. Mark as Done

tituswe committed:
Level-4

tituswe committed:
A-TextUiTesting

tituswe committed:
A-TextUiTesting

tituswe committed:
A-TextUiTesting

tituswe committed:
Level-5

tituswe committed:
Level-6

tituswe committed:
Reorganized Folders

tituswe committed:
Abstracted executable methods to be computed within the Command enums. File organization

tituswe committed:
Implement Level-7. Save

tituswe committed:
Move executable commands from Saturday to Command

tituswe committed:
Modify String fields in Deadline and Event classes to TemporalAccessors

tituswe committed:
Merge branch 'branch-Level-8'

tituswe committed:
Add command to print deadlines/events occuring on a specific date

tituswe committed:
Organize files

tituswe committed:
Add test files to gitignore

tituswe committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

tituswe committed:
Implement A-Gradle. Use Gradle

tituswe committed:
Implement A-MoreOOP

tituswe committed:
Implement A-Packages

tituswe committed:
Implement A-JUnit

tituswe committed:
Create JAR

tituswe committed:
Implement A-Jar

tituswe committed:
Add JavaDocs for Saturday class

tituswe committed:
Implement Level-9. Find

tituswe committed:
Implement A-CodingStandard

tituswe committed:
Add JavaDocs

tituswe committed:
Fix merge conflicts

tituswe committed:
Implement Level-9. Save

tituswe committed:
Implement A-CheckStyle

tituswe committed:
Implement basic GUI with FXML.

tituswe committed:
Bug fixes

swx0 commented on src/main/java/saturday/models/Task.java:
```diff
@@ -0,0 +1,73 @@
+ * @version 0.1
+ */
+public class Task implements Serializable {
+    private boolean done;
```
Boolean variable should sound like booleans.

swx0 commented on src/main/java/saturday/collections/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+            }
+        }
+        return taskListOnDate;
+    }
```
Perhaps adding helper functions for this function can be used to reduce repetition and add clarity.

swx0 reviewed (commented):
Overall good. Only a few minor suggestions proposed.

kalarisng commented on src/main/java/saturday/command/Command.java:
```diff
@@ -0,0 +1,224 @@
+        }
+        throw new SaturdayException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+    }
+}
```
overall very clean code with logical structure!

tituswe committed:
Implement GUI with FXML

tituswe committed:
Implement GUI v2

kalarisng reviewed (commented):
Overall clean code with appropriate JavaDocs!

tituswe committed:
Customize task list

tituswe committed:
Delete unused files

tituswe committed:
GUI classes: refactor into packages

GUI classes (e.g. TaskView, Dashboard) should be abstracted into one
package.

Upon expanding the graphical interface, codebase may become messy.

Let's make sure to package the GUI component files once more components
have been implemented.

Packaging such components will organize the codebase for easier
navigation.

tituswe committed:
Add assertions

tituswe committed:
Code base: improve code quality

Fixed code for easier readability.

Let's refactor the code to use default branching, evoiding empty catch
blocks, deleting dead code and making the happy path prominent.

tituswe committed:
Use streams for storage

tituswe committed:
Merge pull request #2 from tituswe/branch-A-Assertions

Add assertions

tituswe committed:
Merge branch 'master' into branch-A-CodeQuality

tituswe committed:
Merge branch 'master' into branch-A-Streams

tituswe committed:
Merge pull request #3 from tituswe/branch-A-CodeQuality

Branch a code quality

tituswe committed:
Merge branch 'master' into branch-A-Streams

tituswe committed:
Merge pull request #4 from tituswe/branch-A-Streams

Branch a streams

tituswe committed:
Add gradle.yml

tituswe committed:
Implement sort functionality

tituswe committed:
Update README.md

tituswe committed:
Create assets folder

tituswe committed:
Add files via upload

tituswe committed:
Update README.md

tituswe committed:
Update README.md

tituswe committed:
Update README.md

tituswe committed:
Update README.md

tituswe committed:
Update README.md

tituswe committed:
Add files via upload

tituswe committed:
Create assets folder

tituswe committed:
A-UserGuide

tituswe committed:
Fix task list functionality

tituswe committed:
Merge commit '93facb969fb9adc626d1f53268a74fc7b6feef0a'

tituswe committed:
Update README.md

tituswe committed:
Add files via upload

tituswe committed:
Update README.md

tituswe committed:
Update README.md

tituswe committed:
A-UserGuide

tituswe committed:
Update README.md

tituswe committed:
Delete todo.png

tituswe committed:
Update README.md

tituswe committed:
Update README.md

tituswe committed:
Update README.md

tituswe committed:
Update README.md

tituswe committed:
Update README.md

tituswe committed:
Update README.md

tituswe committed:
Update README.md

tituswe committed:
Final version

tituswe committed:
Clean up

tituswe committed:
Update README.md
